compiler/src/dotty/tools/dotc/typer/Inferencing.scala

```diff
@@ -154,15 +157,58 @@ object Inferencing {
    *  their lower bound. Record whether successful.
    *  2nd Phase: If first phase was successful, instantiate all remaining type variables
    *  to their upper bound.
+   *
+   *  Instance types can be improved by replacing covariant occurrences of Nothing
+   *  with fresh type variables, if `force` allows this in its `canImprove` implementation.
    */
   private class IsFullyDefinedAccumulator(force: ForceDegree.Value, minimizeSelected: Boolean = false)
     (using Context) extends TypeAccumulator[Boolean] {
 
-    private def instantiate(tvar: TypeVar, fromBelow: Boolean): Type = {
+    /** Replace toplevel-covariant occurrences (i.e. covariant without double flips)
```

Comment:
Is the special-case here just to be conservative, or is there a reason why we wouldn't want to do the same for double flips (as well as top types in contravariant positions, and lower/upper bounds different from Nothing/Any) ?

Reply:
Just to be conservative